src/Kodeine/Acl/Traits/HasRole.php

```diff
@@ -168,7 +168,7 @@ public function assignRole($role, $model = '', $reference_id = 0)
 
             $roleId = $this->parseRoleId($role);
 
-            if ( ! $this->roles->keyBy('id')->has($roleId) ) {
+            if (count($this->roles()->where("roles.id",$roleId)->wherePivot('model',$model)->wherePivot('reference_id',$reference_id)->get()) == 0)
```

Comment:
Looks like you have forgotten `{`. Fix it and see if there are any issues from Travis.